### PR TITLE
CMake: Move more platform config to platforms/

### PR DESCRIPTION
**Pull request by glebm**
Moves platform libraries, compile definitions, and compile options to the platform CMake files.

### CODE REVIEW

File: CMake/platforms/n3ds.cmake [hunk 1]
```diff
@@ -10,15 +10,20 @@ set(DISABLE_ZERO_TIER ON)
 set(LIBMPQ_FILE_BUFFER_SIZE 32768)
 set(USE_SDL1 ON)
 set(PREFILL_PLAYER_NAME ON)
+set(MO_LANG_DIR \"romfs:/\")
```
Comment (AJenbo) [inline CMake/platforms/n3ds.cmake:13]: This doesn't do anything:
```suggestion
```

File: CMake/platforms/vita.cmake [hunk 2]
```diff
@@ -4,4 +4,14 @@ set(DEVILUTIONX_STATIC_CXX_STDLIB OFF)
 set(BUILD_TESTING OFF)
 set(DISABLE_ZERO_TIER ON)
 set(PREFILL_PLAYER_NAME ON)
-add_definitions("-DMO_LANG_DIR=\"app0:/\"")
+set(MO_LANG_DIR \"app0:/\")
```
Comment (AJenbo) [inline CMake/platforms/vita.cmake:7]:
```suggestion
```

File: CMake/platforms/n3ds.cmake (hunk 1 above)
Comment (glebm) [reply CMake/platforms/n3ds.cmake]: Done

File: CMake/platforms/vita.cmake (hunk 2 above)
Comment (glebm) [reply CMake/platforms/vita.cmake]: Done

**Review by AJenbo — APPROVED** [review]

**Pull request merged**